lib/collection/header.js

```diff
@@ -226,7 +227,14 @@ _.assign(Header, /** @lends Header */ {
             return E;
         }
 
-        return headers.map(Header.unparseSingle).join(separator ? separator : '\n');
+        return headers.reduce(function (acc, header) {
+            // unparse header only if it has a valid key and is not disabled
+            if (header && header.key && !header.disabled) {
```

Comment:
Is blank key a problem? I know … naive question.

Reply:
Yes, it throws `ERR_INVALID_HTTP_TOKEN`

Reply:
@shamasis @kunagpal Issue with Node's HTTP Parser. Works fine with curl and non-NodeJS servers.
I think the best place to handle this specific behavior is in `HeaderList~contentSize`.

Reply:
LGTM otherwise,